services/configuration/src/config_api.rs

```diff
@@ -173,50 +173,43 @@ impl PublicConfigApi {
                 .map(|(cfg_hash, propose_data)| {
                     ApiResponseProposeHashInfo {
                         hash: cfg_hash,
-                        propose_data: propose_data,
+                        propose_data,
                     }
                 })
                 .collect::<Vec<_>>()
         };
-        Ok(proposes)
+        proposes
     }
 
     fn get_all_committed(
         &self,
         previous_cfg_hash_filter: Option<Hash>,
         actual_from_filter: Option<Height>,
-    ) -> Result<Vec<ApiResponseConfigHashInfo>, ApiError> {
+    ) -> Vec<ApiResponseConfigHashInfo> {
         let snapshot = self.blockchain.snapshot();
         let general_schema = Schema::new(&snapshot);
         let index = general_schema.configs_actual_from();
         let committed_configs = {
```

Comment:
It seems like this block has a single statement (a long one, but still). Is this intentional / required by the formatter, or an artifact? IMO, unnecessary indentation should be avoided.

Reply:
This is unfortunately needed to fight the borrow checker.

Reply:
Huh, it seems to work locally for me. I'm talking about the following simple change:

```rust
let committed_configs = index.into_iter()
    .map(|reference| { /* ... */ })
    // ...
    .collect::<Vec<_>>();
committed_configs
```

Btw, there is a similar obsolete block in `get_all_proposes`.

Reply:
Have you tried to run Clippy after that? :wink: 
We have `#[cfg_attr(feature = "cargo-clippy", allow(let_and_return))]` annotations in many places because of this issue. I dislike both variants equally.

Reply:
But even with a block, this still *is* a let-and-return, only not recognized by clippy (yet?). 

In general, do you think iterator lifetimes could be an issue with Exonum? I've not looked into the issue in depth, but it seems that the problem is that the iterators over indexes are constrained by the lifetime of the index (default behavior), whereas they could be constrained by the lifetime of a snapshot / fork.

Reply:
Yes, I suppose, but it should be checked more thoroughly. I has created an [issue](https://jira.bf.local/browse/ECR-737) for that, but I don't know when we will have time to look at that.